misk-aws/src/main/kotlin/misk/dynamodb/RealDynamoDbModule.kt

```diff
@@ -0,0 +1,46 @@
+package misk.dynamodb
+
+import com.amazonaws.auth.AWSCredentialsProvider
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig
+import com.google.inject.Provides
+import misk.cloud.aws.AwsRegion
+import misk.inject.KAbstractModule
+import javax.inject.Singleton
+
+internal class RealDynamoDbModule : KAbstractModule() {
+  override fun configure() {
+    requireBinding<AWSCredentialsProvider>()
+    requireBinding<AwsRegion>()
+  }
+
+  @Provides @Singleton
+  fun providesAmazonDynamoDB(
+    awsRegion: AwsRegion,
+    awsCredentialsProvider: AWSCredentialsProvider
+  ): AmazonDynamoDB {
+    return AmazonDynamoDBClientBuilder
+        .standard()
+        .withRegion(awsRegion.name)
+        .withCredentials(awsCredentialsProvider)
+        .build()
+  }
+
+  @Provides @Singleton
+  fun providesDynamoDBmapper(dynamoDb: AmazonDynamoDB): DynamoDBMapper {
+    return dynamoDbMapper(dynamoDb)
+  }
+
+  companion object {
+    internal fun dynamoDbMapper(dynamoDb: AmazonDynamoDB): DynamoDBMapper {
```

Comment:
inline this function?

Reply:
What's the benefit of that? (newer to kotlin and the syntax is a tad confusing)

Reply:
By inline I mean just removing the function and put the logic in `providesDynamoDBmapper`. Why was it a separate function to start with?

Reply:
Currently it is also used by the Fake module (its in the same package currently )